src/zarr_jpeg/zarr_jpeg.py

```diff
@@ -10,25 +10,38 @@ class jpeg(Codec):
     Parameters
     ----------
     quality : int
-        Compression level. 
+        Compression level.
     """
 
     codec_id = "jpeg"
 
-    def __init__(self, quality=100):
+    def __init__(self, quality=100, axis_reduction='collapse'):
         self.quality = quality
+        self.axis_reduction = axis_reduction
         assert (
             self.quality > 0 and self.quality <= 100 and isinstance(self.quality, int)
         )
         super().__init__()
 
     def encode(self, buf):
         bufa = ensure_ndarray(buf)
-        assert bufa.ndim == 2 or bufa.ndim == 3
-        if bufa.ndim == 3:
-            tiled = bufa.reshape(bufa.shape[0] * bufa.shape[1], -1)
-        else:
+        if isinstance(self.axis_reduction, str) and self.axis_reduction == 'collapse':
+            if bufa.ndim >= 3:
+                # all but the last dimension are collapsed into first dimension, followed by last dimension
+                self.axis_reduction = [[dim for dim in range(bufa.ndim - 1)]]
+                self.axis_reduction.append([bufa.ndim - 1])
+            else:
+                # keep each dimension as is.
+                self.axis_reduction = None
```

Comment:
instead of axis reduction being `None` when no reshaping occurs, can we instead make a valid axis reduction that preserves the original shape? e.g. `[[0],[1]]` for a 2D input. This will avoid an `if axis_reduction is None` check later.

Reply:
Yes, we could.  But we would need the check `if axis_reduction is None:` regardless because `None` could be supplied by the user who wants no axis reduction.  (Unless we want to take that possibility away from the user and require that they supply, e.g., `[[0], [1]]` or `[[0], [1], [2]]` instead.)

Reply:
so what is the difference between `axis_reduction='collapse'` and `axis_reduction=None`?

Reply:
There is no difference for a 2-dimesional array; they both mean the same as `[[0], [1]]`.  For a 3-dimensional array, `collapse` means the same as `[[0, 1], [2]]` and `None` means the same as `[[0], [1], [2]]`

Reply:
makes sense!